### PR TITLE
🎨 Default thumbnails for each product

### PR DESCRIPTION
**Pull request by jsaq007**
## What do these changes do?

* Add a default thumbnail for oSparc, S4L and Tip (Transparent thumbnails visible with empty study)

* Standardise confirmation windows with cancel as text button and always on the left

* Refactor cards on the dashboard allowing maximum 2 lines in both the header and footer for more consistent layout

* Resolves the issue where duplicating studies was not so visible

* Resolves the issue where duplicating studies was not possible in list view

## Related issue/s

- resolves Default thumbnails https://github.com/ITISFoundation/osparc-issues/issues/1212
- resolves https://git.speag.com/oSparc/osparc-s4l/-/issues/722

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/static-webserver/client/source/class/osparc/dashboard/CardBase.js [hunk 1]
```diff
@@ -55,11 +55,12 @@ qx.Class.define("osparc.dashboard.CardBase", {
     MODE_APP: "@FontAwesome5Solid/desktop/13",
     NEW_ICON: "@FontAwesome5Solid/plus/",
     LOADING_ICON: "@FontAwesome5Solid/circle-notch/",
-    STUDY_ICON: "resource/osparc/Thumbnail_Transparent.png",
-    TEMPLATE_ICON: "resource/osparc/Thumbnail_Transparent.png",
-    SERVICE_ICON: "resource/osparc/Thumbnail_Transparent.png",
-    COMP_SERVICE_ICON: "resource/osparc/Thumbnail_Transparent.png",
-    DYNAMIC_SERVICE_ICON: "resource/osparc/Thumbnail_Transparent.png",
+    // Get the default thumbnail for each product else add the image and extension osparc.product.Utils.getProductThumbUrl(Thumbnail-01.png)
+    STUDY_ICON: osparc.product.Utils.getProductThumbUrl(),
```
Comment (pcrespov) [inline services/static-webserver/client/source/class/osparc/dashboard/CardBase.js:59]: Looks much more maintainable this way

**Review by pcrespov — APPROVED** [review]
thx

**Review by elisabettai — APPROVED** [review]
Looks good!  (also discussed this live, I'm curious to see how the transparent oSPARC thumbnail will look like).

Minor: in the description you linked the wrong issue (you added issue #1212 from this repo, but I guess you wanted #osparc-issues#1212 in the osparc-issues repo). I've fixed it for you, so you know for next time. 😉

**Review by GitHK — COMMENTED** [review]
👍

**Review by sanderegg — APPROVED** [review]
pretty nice!

File: services/static-webserver/client/source/class/osparc/dashboard/CardBase.js [hunk 2]
```diff
@@ -522,58 +523,62 @@ qx.Class.define("osparc.dashboard.CardBase", {
...
+        width: 17,
+        padding: 3
       });
-      label.setValue(lbl);
+
+      projectStatusLabel.setVisibility(icn && toolTipText && bdr ? "visible" : "excluded");
+
+      projectStatusLabel.setSource(icn);
+      projectStatusLabel.setToolTipIcon(icn);
+      projectStatusLabel.setToolTipText(toolTipText);
```
Comment (odeimaiz) [inline services/static-webserver/client/source/class/osparc/dashboard/CardBase.js:581]:
```suggestion
      projectStatusLabel.set({
        visibility: icn && toolTipText && bdr ? "visible" : "excluded",
        source: icn,
        toolTipIcon: icn,
        toolTipText
      });
```

File: services/static-webserver/client/source/class/osparc/dashboard/GridButtonBase.js [hunk 3]
```diff
@@ -36,38 +37,14 @@ qx.Class.define("osparc.dashboard.GridButtonBase", {
...
-    });
-    mainLayout.setLayout(grid);
+    const mainLayout = this.getChildControl("main-layout");
 
     this._add(mainLayout, {
       top: 0,
       right: 0,
       bottom: 0,
       left: 0
     });
```
Comment (odeimaiz) [inline services/static-webserver/client/source/class/osparc/dashboard/GridButtonBase.js:47]: I would move these lines to the ``_createChildControlImpl``

**Review by odeimaiz — APPROVED** [review]
Awesome 💅! Thanks a lot  :)

File: services/static-webserver/client/source/class/osparc/dashboard/ListButtonPlaceholder.js [hunk 4]
```diff
@@ -26,31 +26,60 @@ qx.Class.define("osparc.dashboard.ListButtonPlaceholder", {
     // make unselectable
     this.addListener("changeValue", () => this.setValue(false), this);
 
+    this.__layout = this.getChildControl("progress-layout")
```
Comment (odeimaiz) [inline services/static-webserver/client/source/class/osparc/dashboard/ListButtonPlaceholder.js:29]: I don't think you need to create the private member here. Just ``this.getChildControl("progress-layout")`` when you want to add the subitems to the layout, same as in the ``GridButtonBase``

**Review by odeimaiz — APPROVED** [review]
Very nice, thanks for the effort 👌

**Review by elisabettai — APPROVED** [review]
Looks good, thanks a lot also for fixing the duplication UI colors.

Only thing to double check is that the duplicated projects/studies are still at the top of the list (OM has fixed this a couple of days ago).

**Pull request merged**